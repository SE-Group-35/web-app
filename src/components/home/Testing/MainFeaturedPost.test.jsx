import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MainFeaturedPost from '../MainFeaturedPost';
import CheckPropTypes from 'check-prop-types';
import { ThemeProvider, makeStyles } from '@material-ui/core/styles';
import image from "../../../assets/images/homecover4.jpg";


configure({adapter: new Adapter()});

describe('<EventCard />', () => {

    let wrapper;
    const theme = {} ;

    beforeEach(() => {       
        wrapper = shallow(<ThemeProvider theme={theme}><MainFeaturedPost post={mainFeaturedPost}/></ThemeProvider>);
    });
       
    const mainFeaturedPost = {
        title: "Let's Make Your Best Trip Ever",
        description:
          "Hi! Would you explore nature paradise in the world. Let's find the best trip in Sri Lanka.Join with us!.",
        image: `${image}`,
        imgText: "main image description",
      };
    const wrongPropType ={
          post:3
    };

    it('should not throw an error if correct prop types are passed', () => {
        
        const propsErr = CheckPropTypes(MainFeaturedPost.propTypes,mainFeaturedPost,'props',MainFeaturedPost.name);
        expect(propsErr).toBeUndefined();
    });
    it('should throw an error if incorrect prop types are passed', () => {        
        const propsErr = CheckPropTypes(MainFeaturedPost.propTypes,wrongPropType,'props',MainFeaturedPost.name);        
        expect(propsErr).toBeDefined();
    }); 

    it('should matches the snapshot',() => {
        expect(wrapper.getElements()).toMatchSnapshot();
    });
    
    
         
    
    
});