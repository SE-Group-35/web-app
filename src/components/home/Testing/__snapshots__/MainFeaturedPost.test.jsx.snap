// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventCard /> should matches the snapshot 1`] = `
Array [
  <ThemeProvider
    theme={Object {}}
  >
    <InnerThemeProvider>
      <MainFeaturedPost
        post={
          Object {
            "description": "Hi! Would you explore nature paradise in the world. Let's find the best trip in Sri Lanka.Join with us!.",
            "image": "homecover4.jpg",
            "imgText": "main image description",
            "title": "Let's Make Your Best Trip Ever",
          }
        }
      />
    </InnerThemeProvider>
  </ThemeProvider>,
]
`;
